db/view/view.cc

```diff
@@ -1603,22 +1603,23 @@ future<stop_iteration> view_update_builder::on_results() {
 }
 
 view_update_builder make_view_update_builder(
-        data_dictionary::database db,
+        replica::database& db,
         const replica::table& base_table,
         const schema_ptr& base,
         std::vector<view_and_base>&& views_to_update,
         mutation_reader&& updates,
         mutation_reader_opt&& existings,
         gc_clock::time_point now) {
     auto vs = views_to_update | std::views::transform([&] (view_and_base v) {
+        auto holder = db.find_column_family(v.view).view_build_gate().hold();
         if (base->version() != v.base->base_schema()->version()) {
             on_internal_error(vlogger, format("Schema version used for view updates ({}) does not match the current"
                                               " base schema version of the view ({}) for view {}.{} of {}.{}",
                 base->version(), v.base->base_schema()->version(), v.view->ks_name(), v.view->cf_name(), base->ks_name(), base->cf_name()));
         }
-        return view_updates(std::move(v));
+        return view_updates(std::move(v), std::move(holder));
```

Comment:
Note that `make_view_update_builder` is also used for regular view updates, so this will also hold the gate for every regular view update. I think we should try to move the gate somewhere more specific to view building or make sure that this `view_update_builder` is for view building

Reply:
> perhaps it makes sense to stop also regular view updates? I suppose a view update that's running while the view is removed could be exposed to similar issues, although I'm not sure if it happens in practice

I believe it make sense. If the gate is closed then we are dropping the view so there is no use in applying any more view updates, regular or not.